### DISCUSSION TITLE
Consistent Timeouts After "1062" Duplicate Entry Errors

### DISCUSSION THREAD

**Discussion by marwan-at-work** [planetscale/discussion#27] [Bug Reports] [answered]
I have noticed that whenever I get a uniqueness constraints error, the _following_ INSERT/DELETE queries always get timed out. 

To reproduce: 

1. Create a `users` table with a unique column called "email"
2. Create a Go db using @fatih 's answer: https://github.com/planetscale/beta/discussions/24#discussioncomment-759432
2. Run the following Go code: 

```golang
func main() {
	// this works
	tx, err := db.Begin()
	must(err)
	_, err = tx.Exec(`INSERT INTO users (email) values ('one')`)
	must(err)
	must(tx.Commit())
	
	
	// force a duplicate entry error
	tx, err = db.Begin()
	must(err)
	_, err = tx.Exec(`INSERT INTO users (email) values ('one')`)
	if err == nil {
		must(fmt.Errorf("expected a duplicate entry error"))
	} else {
		// rollback the transaction
		fmt.Println(tx.Rollback())
	}
	
	// This is where the bug is:
	tx, err = db.Begin()
	must(err)
	_, err = tx.Exec(`DELETE FROM users`)
	must(err) // the error happens here after a 20 second timeout
	must(tx.Commit())
}

func must(err error) {
	if err != nil {
		panic(err)
	}
}
```

I end up with the following error: 

> Error 1317: target: gofigdev.-.master: vttablet: (errno 2013) due to context deadline exceeded, elapsed time: 20.000052422s, killing query ID 26989 (CallerID: unsecure_grpc_client) [recovered]

**Comment by aquarapid** [comment]
Verified manually (using `begin/commit/rollback`);  I did not run into the timeout;  but I do see a 15 second or so pause after executing the `DELETE from users`;  you probably hit the timeout because you do this programmatically.

Our engineering team is looking into it.

**Comment by mattrobenolt** [comment, marked as answer]
This should be fixed as well. Thanks for the report!

**Reply by marwan-at-work** [reply]
Can confirm 🙂
